sp/src/game/shared/hl2/basehlcombatweapon_shared.cpp

```diff
@@ -412,6 +456,130 @@ Activity CBaseHLCombatWeapon::ActivityOverride( Activity baseAct, bool *pRequire
+{
+	m_iClip1 /= 2;
+	m_iClip2 /= 2;
+
+	CBaseCombatCharacter *pOwner = GetOwner();
+
+	BaseClass::Drop( vecVelocity );
+
+	if (pOwner && pOwner->IsPlayer())
+	{
+		// Spawn left hand gun when regular one is dropped
+		if (GetLeftHandGun())
+		{
+			// Drop the fake gun
+			CBaseEntity *pRealGun = CreateNoSpawn( GetClassname(), GetAbsOrigin() - Vector( 0, 0, BoundingRadius() ), GetAbsAngles(), this );
+			DispatchSpawn( pRealGun );
+
+			if (pRealGun)
+			{
+				if (pRealGun->VPhysicsGetObject())
+				{
+					// Copy velocity
+					Vector vecVelocity;
+					AngularImpulse vecAngVelocity;
+					VPhysicsGetObject()->GetVelocity( &vecVelocity, &vecAngVelocity );
+					pRealGun->VPhysicsGetObject()->SetVelocity( &vecVelocity, &vecAngVelocity );
+				}
+
+				pRealGun->MyCombatWeaponPointer()->m_iClip1 = m_iClip1;
+				pRealGun->MyCombatWeaponPointer()->m_iClip2 = m_iClip2;
+
+				// Act as if it was dropped the same way
+				pRealGun->SetThink( &CBaseCombatWeapon::SetPickupTouch );
+				pRealGun->SetTouch( NULL );
+				pRealGun->SetNextThink( gpGlobals->curtime + 1.0f );
+			}
+
+			UTIL_Remove( GetLeftHandGun() );
+
+			SetLeftHandGun( NULL );
+		}
+	}
+}
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+CBaseAnimating *CBaseHLCombatWeapon::CreateLeftHandGun()
+{
+	if (!GetOwner())
+	{
+		Warning( "%s cannot create left hand gun because it has no owner\n", GetDebugName() );
+		return NULL;
+	}
+
+	if (!CanDualWield())
+	{
+		Warning( "%s not capable of dual wielding\n", GetDebugName() );
+		return NULL;
+	}
+
+	if (GetLeftHandGun())
+	{
+		Warning("%s already has a left-handed weapon (%s)\n", GetDebugName(), GetLeftHandGun()->GetDebugName());
+		return NULL;
+	}
+
+	// Create a fake second gun
+	CBaseEntity *pEnt = CBaseEntity::CreateNoSpawn( "prop_dynamic_override", this->GetLocalOrigin(), this->GetLocalAngles(), this );
+	if (pEnt)
+	{
+		char szLeftModel[MAX_PATH];
+		if (GetWpnData().szWorldModelDual[0])
+		{
+			V_strncpy( szLeftModel, GetWpnData().szWorldModelDual, sizeof( szLeftModel ) );
+		}
+		else
+		{
+			// If there is no specific worldmodel, then just try adding "_left" to the end of the model name
+			V_StripExtension( GetWorldModel(), szLeftModel, sizeof( szLeftModel ) );
+			V_strncat( szLeftModel, "_left.mdl", sizeof( szLeftModel ) );
+		}
+
+		pEnt->SetModelName( MAKE_STRING( szLeftModel ) );
+		pEnt->SetRenderMode( kRenderTransColor );
+		DispatchSpawn( pEnt );
+		pEnt->FollowEntity( GetOwner(), true );
+		pEnt->SetOwnerEntity( this );
+		pEnt->AddEffects( EF_NOSHADOW );
+
+		CBaseAnimating *pAnimating = static_cast<CBaseAnimating *>(pEnt);
+		SetLeftHandGun( pAnimating );
+
+		if (GetOwner()->IsPlayer())
+		{
+			SetModel( GetViewModel() );
+
+			if (GetOwner()->GetActiveWeapon() == this)
+			{
+				// Play the first draw animation and re-deploy the model
+				m_bFirstDraw = true;
```

Comment:
What if the weapon is set to not first draw? (Without checking, I can't remember if m_bFirstDraw respects those checks or not)

Reply:
`m_bFirstDraw` indeed respects those checks on its own